### PR TITLE
kubelet: cleanup pkg/kubelet/leaky package

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Package `leaky` holds bits of kubelet that should be internal but have leaked out through bad abstractions.
This PR tries to clean it up.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
/assign @bobbypage 

@bobbypage can you take a look? Does it make code better?

File: pkg/kubelet/leaky/leaky.go [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-*/
-
-// Package leaky holds bits of kubelet that should be internal but have leaked
-// out through bad abstractions.  TODO: delete all of this.
-package leaky
-
-const (
-	// PodInfraContainerName is used in a few places outside of Kubelet, such as indexing
-	// into the container info.
-	PodInfraContainerName = "POD"
```
Comment (haircommander) [inline pkg/kubelet/leaky/leaky.go:24]: FWIW: CRI-O uses this field as well. I am open to moving it, but I think it should be exported. Probably also worth a comment that's why it's exported

Comment (SataQiu) [reply pkg/kubelet/leaky/leaky.go:24]: Thanks for your feedback @haircommander 
According to the package comment, it seems that this field should not be exposed.
But if this field is already used by other projects, maybe we should move it to a more reasonable location, such as `k8s.io/kubelet`. 

What do you think? @SergeyKanzhelev @bobbypage

Comment (endocrimes) [reply pkg/kubelet/leaky/leaky.go:24]: The unfortunate part of this is that it already leaked out of the codebase - and so just removing it isn't something we can reasonably do. 

Removing the package doesn't really fix the abstraction, but removes the bandage :(

Comment (SataQiu) [reply pkg/kubelet/leaky/leaky.go:24]: Agree, we need to evaluate how to deal with the leaked field again.

**Comment by SataQiu** [comment]
> Removing the package doesn't really fix the abstraction, but removes the bandage

/close

**Pull request closed without merging**